pkg/ds/daemon_set.go

```diff
@@ -97,14 +96,15 @@ func (ds *daemonSet) WatchDesires(
 	quitCh <-chan struct{},
 	updatedCh <-chan *fields.DaemonSet,
 	deletedCh <-chan *fields.DaemonSet,
-	nodesChangedCh <-chan struct{},
 ) <-chan error {
 	errCh := make(chan error)
+	nodesChangedCh := make(chan struct{})
```

Comment:
Why not `nodesChangedCh := ds.watchNodeChanges(quitCh)` and remove the goroutine you added to the end of the method? Would that accomplish the same thing?